### ISSUE TITLE
Invalid response for ?requestPayment sub-resource request on Bucket

### ISSUE THREAD

**Issue by fviard** [minio/minio#9394]
## Current behavior
When doing a request to the sub-resource ?requestPayment of a bucket, (Ex.: `mybucket?requestPayment`), Minio response has the status code 200 (Ok) and an empty response data.

See an example of reply from Minio:
```
DEBUG: format_uri(): /boom/?requestPayment
DEBUG: Sending request method_string='GET', uri=u'/boom/?requestPayment', headers={'x-amz-content-sha256': u'e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855', 'Authorization': u'AWS4-HMAC-SHA256 Credential=/20200420/us-east-1/s3/aws4_request,SignedHeaders=host;x-amz-content-sha256;x-amz-date,Signature=adc7d4c4d19b0621c60819671edcdbf6656485cde132f83a2814ba37d514166e', 'x-amz-date': '20200420T140300Z'}, body=(0 bytes)
DEBUG: ConnMan.put(): connection put back to pool (http://127.0.0.1:9000#4)
DEBUG: Response:
{'data': '',
 'headers': {'accept-ranges': 'bytes',
             'content-length': '0',
             'content-security-policy': 'block-all-mixed-content',
             'date': 'Mon, 20 Apr 2020 14:03:00 GMT',
             'server': 'MinIO/RELEASE.2020-03-14T02-21-58Z',
             'vary': 'Origin',
             'x-amz-request-id': '16078C0CEDCC8049',
             'x-xss-protection': '1; mode=block'},
 'reason': 'OK',
 'status': 200}
```

## Expected behavior
This is an incorrect behavior.
**If Minio does not support this feature or has nothing to send as reply, it should send a response with one the followings errors/status codes:**

> * 404 Not Found
> * 501 Not Implemented

## Steps to Reproduce (for bugs)
1) (Create a bucket) s3cmd -c s3cfg_localminio mb s3://boom
2)  (Look at the reply for the ?requestPayment request) s3cmd -c s3cfg_localminio info s3://boom -d

More logs:
 https://github.com/s3tools/s3cmd/issues/1093

## Regression
Is this issue a regression?  No 

## Your Environment
minio version RELEASE.2020-03-14T02-21-58Z
Linux

**Comment by harshavardhana** [comment]
Request payment only makes sense for public cloud, this is not useful in MinIO so it never going to be implemented @fviard 

We return back 200 OK because some applications simply do a GET request without doing any future PUT operations. There is no good reason for us to return 501 as we do not implement PUT.

**Comment by fviard** [comment]
Again, the question is not if you want to support the feature or not.
When a sub-ressource is requested,
- either you reply with some content that is a completely FILLED XML with the mandatory fields
- Or you use HTTP to reply with 404 or 501 to indicate that there is not such a sub-resource, or that you don't support that sub-resource.

So that you understand correctly, if I ask for mybucket?anyresource,
a 404 reply does not mean that "mybucket" does not exist, but just that mybucket does not have the "anyresource" ressource.

It is the same think if you support versions one day, if I ask you myobject?version=14.
if there is no version, or not such a version, you should reply 404, and not an empty content...

**Comment by harshavardhana** [comment]
@fviard the point here is no that some applications behave differently here we cannot change that right now.

**Comment by fviard** [comment]
@harshavardhana I have hard time to understand if you read too fast my issues or what, but you are not very clear on what is the problem.

For the moment you have 2 bugs, that and #9396 .
ie, for both of them, you are not behaving like amazon s3 and not respecting the s3 protocol.
Sorry if it is rude, but for these things, Minio is sending back garbage.
(Like for the CORS, you are sending a non-sense part of xml...)

And again, THIS IS NOT RELATED to the fact to not support both features, that is totally acceptable.

You can't say that you can't change for other "clients" legacy behavior, because all the clients are working with AWS that does exactly what I tell you.

That you tell me that it is not a priority, I can understand, it is not my problem, but that you put them on "wontfix", saying that it is completely fine, I'm a little bit disappointed.
I used to like Minio that was reactive, and doing the right thing, but here, I can't understand your attitude.
Again, I don't ask you to support feature, just to having the server send a correct response for unsupported features and not garbage...

**Comment by harshavardhana** [comment]
> @harshavardhana I have hard time to understand if you read to fast me issues or what, but you are not very clear on what is the problem.
> 
> For the moment you have 2 bugs, that and #9396 .
> ie, for both of them, you are not behaving like amazon s3 and not respecting the s3 protocol.
> Sorry if it is rude, but for these things, Minio is sending back garbage.
> (Like for the CORS, you are sending a non-sense part of xml...)
> 
> And again, THIS IS NOT RELATED to the fact to not support both features, that is totally acceptable.
> 
> You can't say that you can't change for other "clients" legacy behavior, because all the clients are working with AWS that does exactly what I tell you.
> 
> That you tell me that it is not a priority, I can understand, it is not my problem, but that you put them on "wontfix", saying that it is completely fine, I'm a little bit disappointed.
> I used to like Minio that was reactive, and doing the right thing, but here, I can't understand your attitude.
> Again, I don't ask you to support feature, just to having the server send a correct response for unsupported features and not garbage...

I am telling it's not possible for us to address this right now.

**Comment by fviard** [comment]
Ok, so you remove the "wontfix" and just put it on stack of bugs to fix "when possible", right?

**Comment by harshavardhana** [comment]
> Ok, so you remove the "wontfix" and just put it on stack of bugs to fix "when possible", right?

Ah that I can..

**Comment by fviard** [comment]
Perfect, thank you very much, that was all that I was asking, there is no emergency, :-)

**Comment by harshavardhana** [comment]
> @harshavardhana I have hard time to understand if you read too fast my issues or what, but you are not very clear on what is the problem.
> 
> For the moment you have 2 bugs, that and #9396 .
> ie, for both of them, you are not behaving like amazon s3 and not respecting the s3 protocol.
> Sorry if it is rude, but for these things, Minio is sending back garbage.
> (Like for the CORS, you are sending a non-sense part of xml...)
> 
> And again, THIS IS NOT RELATED to the fact to not support both features, that is totally acceptable.
> 
> You can't say that you can't change for other "clients" legacy behavior, because all the clients are working with AWS that does exactly what I tell you.
> 

You are right about this but there are applications which we cannot change which rely on this behavior. It may be but we don't have the luxury to fix those applications.

**Comment by fviard** [comment]
You mean one of yours that you don't have the resources for the moment to change, or external one that you have no control over?

**Comment by harshavardhana** [comment]
> You mean one of yours that you don't have the resources for the moment to change, or external one that you have no control over?

Anything that we have control over we can change, all the current changes are there for reasons for things which we don't have control over.

**Comment by harshavardhana** [comment]
This is fixed in PR #9586
